pallets/pooled-staking/src/lib.rs

```diff
@@ -0,0 +1,481 @@
+            released: T::Balance,
+        },
+        RequestedUndelegate {
+            candidate: Candidate<T>,
+            delegator: Delegator<T>,
+            from: TargetPool,
+            pending: T::Balance,
+            released: T::Balance,
+        },
+        ExecutedUndelegate {
+            candidate: Candidate<T>,
+            delegator: Delegator<T>,
+            released: T::Balance,
+        },
+
+        /// Stake of that Candidate increased.
+        IncreasedStake {
+            candidate: Candidate<T>,
+            stake_diff: T::Balance,
+        },
+        /// Stake of that Candidate decreased.
+        DecreasedStake {
+            candidate: Candidate<T>,
+            stake_diff: T::Balance,
+        },
+        /// Delegator staked towards a Candidate for AutoCompounding Shares.
+        StakedAutoCompounding {
+            candidate: Candidate<T>,
+            delegator: Delegator<T>,
+            shares: T::Balance,
+            stake: T::Balance,
+        },
+        /// Delegator unstaked towards a candidate with AutoCompounding Shares.
+        UnstakedAutoCompounding {
+            candidate: Candidate<T>,
+            delegator: Delegator<T>,
+            shares: T::Balance,
+            stake: T::Balance,
+        },
+        /// Delegator staked towards a candidate for ManualRewards Shares.
+        StakedManualRewards {
+            candidate: Candidate<T>,
+            delegator: Delegator<T>,
+            shares: T::Balance,
+            stake: T::Balance,
+        },
+        /// Delegator unstaked towards a candidate with ManualRewards Shares.
+        UnstakedManualRewards {
+            candidate: Candidate<T>,
+            delegator: Delegator<T>,
+            shares: T::Balance,
+            stake: T::Balance,
+        },
+        /// Collator has been rewarded.
+        RewardedCollator {
+            collator: Candidate<T>,
+            auto_compounding_rewards: T::Balance,
+            manual_claim_rewards: T::Balance,
+        },
+        /// Delegators have been rewarded.
+        RewardedDelegators {
+            collator: Candidate<T>,
+            auto_compounding_rewards: T::Balance,
+            manual_claim_rewards: T::Balance,
+        },
+        /// Rewards manually claimed.
+        ClaimedManualRewards {
+            candidate: Candidate<T>,
+            delegator: Delegator<T>,
+            rewards: T::Balance,
+        },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        InvalidPalletSetting,
+        DisabledFeature,
+        NoOneIsStaking,
+        StakeMustBeNonZero,
+        RewardsMustBeNonZero,
+        MathUnderflow,
+        MathOverflow,
+        NotEnoughShares,
+        TryingToLeaveTooSoon,
+        InconsistentState,
+        UnsufficientSharesForTransfer,
+        CandidateTransferingOwnSharesForbidden,
+        RequestCannotBeExecuted(u16),
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        pub fn rebalance_hold(
```

Comment:
Shouldn't extrinsics have call_index and weight?

Reply:
Currently the pallet is in `dev_mode`, which allows to skip adding the weights. This should be removed and benchmark should be run.

Reply:
I can remove the dev_mode there